src/daemon/rpmostreed-daemon.c

```diff
@@ -616,15 +641,56 @@ rpmostreed_get_client_uid (RpmostreedDaemon *self,
   return TRUE;
 }
 
+static gboolean
+get_client_pid (RpmostreedDaemon *self,
+                const char       *client,
+                pid_t            *out_pid)
+{
+  g_autoptr(GError) local_error = NULL;
+  g_autoptr(GVariant) all = g_dbus_proxy_call_sync (self->bus_proxy,
+                                                    "GetConnectionUnixProcessID",
+                                                    g_variant_new ("(s)", client),
+                                                    G_DBUS_CALL_FLAGS_NONE,
+                                                    2000, NULL, &local_error);
+  if (!all)
+    {
+      sd_journal_print (LOG_WARNING, "Failed to GetConnectionUnixProcessID for client %s: %s",
+                        client, local_error->message);
+      return FALSE;
+    }
+
+  g_variant_get (all, "(u)", out_pid);
+  return TRUE;
+}
+
+/* Given a DBus address, load metadata for it */
+static struct RpmOstreeClient *
+client_new (RpmostreedDaemon *self, const char *address)
+{
+  struct RpmOstreeClient *client = g_new0 (struct RpmOstreeClient, 1);
+  client->address = g_strdup (address);
+  if (rpmostreed_get_client_uid (self, address, &client->uid))
+    client->uid_valid = TRUE;
+  if (get_client_pid (self, address, &client->pid))
+    {
+      client->pid_valid = TRUE;
+      if (sd_pid_get_user_unit (client->pid, &client->sd_unit) == 0)
```

Comment:
So in your example, this returned `session-1.scope` for the cli. Would it (or the following call) return e.g. `rpm-ostreed-automatic.service` for the automatic updates timer?

Reply:
Yep with this it looks like:

`May 16 16:18:56 localhost.localdomain rpm-ostree[3390]: client(id:cli dbus:1.47 unit:rpm-ostreed-automatic.service uid:0) vanished; remaining=0`

And now checked in a test :arrow_down: